### ISSUE TITLE
E7003 Errors when using Fn::Transform inside a Mapping

### ISSUE THREAD

**Issue by speshak** [aws-cloudformation/cfn-lint#2016]
*cfn-lint version: 0.49.2*

*Description of issue.*
#2006 tightened what is considered valid for use in a Mapping.  This causes it to reject what otherwise appears to be a valid use of `Fn::Transform` as the body of a Mapping.

For example, this snippet is valid CFN:

```yaml
Mappings:
  AwsAgentPlatformMap:
    Fn::Transform:
      Name: AWS::Include
      Parameters:
        Location: s3://my-bucket-name/version/3.0.1/amazonlinux2/a-json-file.json
```

This usage trips the newly enhanced regex:

```
E7003 Mapping key (Fn::Transform) has invalid name. Name has to be alphanumeric, '-' or '.'
```

**Comment by PatMyron** [comment]
see https://github.com/aws-cloudformation/cfn-lint/issues/2014, planning on continuing to track this in https://github.com/aws-cloudformation/cfn-lint/issues/476
